src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/ExceptionHandling.cs

```diff
@@ -664,10 +787,12 @@ private static void DispatchEx(scoped ref StackFrameIterator frameIter, ref ExIn
                         catchingTryRegionIdx = MaxTryRegionIdx;
                     }
                 }
+#endif // FEATURE_OBJCMARSHAL && NATIVEAOT
+                handlingFrameSP = frameIter.SP;
+                catchingTryRegionIdx = MaxTryRegionIdx;
             }
-#endif // FEATURE_OBJCMARSHAL
 
-            if (pCatchHandler == null && pReversePInvokePropagationCallback == IntPtr.Zero)
+            if (pCatchHandler == null && pReversePInvokePropagationCallback == IntPtr.Zero && !unwoundReversePInvoke)
```

Comment:
```suggestion
            if (pCatchHandler == null && pReversePInvokePropagationCallback == IntPtr.Zero
#if !NATIVEAOT
                && !unwoundReversePInvoke
#endif
                )
```

Currently NativeAot fails fast here if there is no `pCatchHandler` or `pReversePInvokePropagationCallback`. If I'm reading the current state of  this PR correctly, if this change would cause the NativeAot to pass the null `pCatchHandler` to `RhpCallCatchFunclet` where it would try to jump to the null pointer. I'm pretty sure this would prevent NativeAot from printing the nice managed stack trace on an unhandled exception.

Reply:
@AustinWise thank you for spotting this! You are right that the new part of the condition is meant only for the non-native aot case. I was just investigating why the nativeaot exception handling test was failing in this PR, so you've saved me some valuable time!